Import all sprites in the project

Tristramg opened the pull request:
Ref #10627

## What does this pull request do

This pull requests imports the sprites to have them locally instead of a remote server (4f6659c5530beadbcf84433c0ad0c58d0501903a)

We also change the tool to generate. This reduces a bit the size of the generated atlas and is — arguably — a little bit easier to use 92a75ff00296c154c8ab8fc006516d1bc8ae664e

The front now uses those new sprites f60ec385df90aa988b46f1dc9aa5c1f2df13411e

## Benefits

* Less dependency to an external server
* Smaller files (~1Mb saved per map view)

## How to test

There should be no behavior change.

Navigate through the map with the signal layer on. See in the console if there are no missing sprites and in the network tools that all `sprites` files are correctly loaded from the osrd server and no longer https://static.osm.osrd.fr/sprites/sprites

In editoast, run generate-atlas.sh (it shouldn’t change anything, but output what atlases have been generated). This requires spreet (`cargo install spreet`)

Tristramg committed:
editoast: assets: import signal svg from nicolaswurtz/signalisation-rfn-svg

Signed-off-by: Tristram Gräbener <tristram+git@tristramg.eu>

Tristramg committed:
editoast: assets: sprites: adapt generate-atlas to use spreet and search in subdirectories

Signed-off-by: Tristram Gräbener <tristram+git@tristramg.eu>

Tristramg committed:
front: map: use the sprites for signals served by editoast

This reduces a dependency to an external server

The sprites are also smaller
0.41Mb instead of 1.2Mb for the default sprite @2x

Signed-off-by: Tristram Gräbener <tristram+git@tristramg.eu>

emersion commented:
How large are these new assets?

Tristramg commented:
When taking the largest on my computer (@2x, I suppose this would be the same for most users), that used to be hosted separately
- Before : 1.28 Mb
- After : 409 kb

You can see in the diff on github the size reduction for the existing sprites (~40%)

flomonster reviewed (changes requested):
You are using way too much sprites.
For example all "Arret *" are not used at all (referenced only in `SignalsName.ts`).

It's annoying to put all sprites under `signal_sprites` when some are not signals (e.g. “HEURTOIR”). 
- We could rename this folder to `sprites.
- We could move the default sprites in another folder.

Tristramg committed:
editoast: sprites: rename signal_sprites to sprites

We added signals that are no longer only for signals

Signed-off-by: Tristram Gräbener <tristram+git@tristramg.eu>

Tristramg commented:
Thank you for the feedback

1. I move `signal_sprites` to `sprites` (folder with the assets and its path in `sprites.rs`) : 671e696e
2. Instead of just dumping all the icons of the previous sprite, I looked in the code : 49bb8e29
  *  grep `icon-image`
    * MATD
    * MATG
    * HEURTOIR
    * UNKNOWN2 (for holyland)
    * UNKNOWN
    * TIV D FIXE *
    * TIVD B FIX *
    * R
    * Z
    * RIGHT
  * select distinct(data->'type') from infra_layer_neutral_sign;
    * CC FIN 
    * REV 
    * BP DIS
    * CC EXE
    * BP EXE
    * SECT (this one is missing in the blueprint)
    * BP FIN
  * Actually the blue icons are used in the blueprint style for neutral_sign. Sorry for that, I added them
  * I just kept used images and their blue counterpart
 
As a result, the served PNG is now less than 300kb big, and of course we need less space in the repo

flomonster commented on editoast/assets/sprites/default/SECT.svg:
Where this is used?

thibautsailly replied:
I'm not aware of a real usage today for this display.

Tristramg replied:
If I understand the code correctly, it might be used in https://github.com/OpenRailAssociation/osrd/blob/dev/front/src/common/Map/Layers/InfraObjectLayers/extensions/SNCF/NeutralSectionSigns.tsx#L23
Where type comes from the table `infra_layer_neutral_sign` where SECT exists

flomonster reviewed (approved):
LGTM, I wonder if we want to keep the blueprint style. It was there to visualize the infra schematic (which no longer exists).

We should ask @thibautsailly. If it's not essential to keep this style, we could drop all the blue sprites.

woshilapin commented on editoast/assets/sprites/generate-atlas.sh:
```diff
@@ -1,4 +1,8 @@
+#!/bin/bash
+
+# Allows to use **/*.svg to find files in the directory and subdirectory
+# Requires at leas bash4, release in 2009
```
```suggestion
# Requires at least bash4, release in 2009
```

woshilapin commented on editoast/assets/sprites/generate-atlas.sh:
```diff
@@ -1,4 +1,8 @@
-#!/bin/sh
```
Maybe we want to stay POSIX compatible? If I understand correctly, the move to `bash` is about the `globstar` option? You might be able to change the `cp` command with something along these lines:

```sh
find "${signaling_system}" -type f -name "*.svg" -exec cp "{}" "${svg_dir}" ";"
```

Tristramg replied:
Sorry, I got mixed up in my versions.

By switching to spreet, we now have smaller png, and it can crawl subdirectories, so no need to switch to 15 year young stuff, and no need for a syntax that always needs to be googled ;)

woshilapin commented on editoast/assets/sprites/default/sprites@3x.png:
Since this file is generated from the other files (if I understand correctly), should it be commited? Maybe it can be generated in the `Dockerfile` instead?

Tristramg replied:
Until now, the sprites where committed. We can rethink the existing way to do it.
What about people not using Docker when running editoast? How can we make sure that they won’t blame the front when some signals are missing on the front?

woshilapin replied:
If I understand correctly, it's a deployment problem. And the only way we document how to deploy `editoast` for now is through `docker` (and when I mean documentation, I mostly mean a `Dockerfile` and `osrd-chart`... so mostly code). So I would say that this is ok to make it part of the `Dockerfile`. We could also add a section in some `README`.

Tristramg replied:
All right, so I added this commit: b96f98b333d06d16d101f3c97f683dd3c67d8694 to generate the sprites instead of comiting them.
This should make @emersion  happy to have less data in the repository

@flomonster do you have an opinion?

If every body agrees, I will squash the commit to not have the generated sprite in the history and I think we would be good to go

flomonster replied:
Generating sprites during the docker build is a good idea. The drawback is for editoast developers. When we run it without docker, it doesn't generate the static files.

Tristramg replied:
Yes, I added the information to the readme, but some might still be surprised
I don’t think there is a silverbullet here. Maybe we could add a justfile at some point that runs everything, but that’s clearly far from the scope of this pull request

SharglutDev reviewed (approved):
Lgtm and tested, sprites are still there :) (except the remaining error for etcs)

Tristramg committed:
editoast: assets: do not commit sprites assets

They are now built either during the docker build
Or must be generated by hand as described in the readme

Signed-off-by: Tristram Gräbener <tristram+git@tristramg.eu>

Tristramg committed:
editoast: assets: add ETCS icon

The source is https://github.com/Nakaner/railway-signals/

The licence is CC-0, so we are good

Signed-off-by: Tristram Gräbener <tristram+git@tristramg.eu>
